torch_frame/transforms/mutual_information_sort.py

```diff
@@ -54,9 +54,13 @@ def _fit(self, tf_train: TensorFrame, col_stats: Dict[str, Dict[StatType,
             raise ValueError("The transform can be only used on TensorFrame"
                              " with numerical only features.")
         feat_train = tf_train.feat_dict[stype.numerical]
+        y_train = tf_train.y
         if torch.isnan(feat_train).any():
             feat_train = self._replace_nans(feat_train, self.na_strategy)
-        mi_scores = self.mi_func(feat_train.cpu(), tf_train.y.cpu())
+        if torch.isnan(tf_train.y).any():
+            y_train = self._replace_nans(y_train.view(-1, 1),
+                                         self.na_strategy).squeeze(1)
+        mi_scores = self.mi_func(feat_train.cpu(), y_train.cpu())
```

Comment:
if `y` is nan, shouldn't we just drop the entry on both y and feat sides, rather than applying NA strategy?

Reply:
Ohh. That makes sense~